client/test/build.gradle

```diff
@@ -27,7 +26,7 @@ targetCompatibility = JavaVersion.VERSION_1_7
 sourceCompatibility = JavaVersion.VERSION_1_7
 
 dependencies {
-  compile "org.apache.httpcomponents:httpcore:${versions.httpcore}"
+  compile "org.elasticsearch.client:elasticsearch-rest-client:${version}"
```

Comment:
I think I get it. either we use the ordinary httpcore classes, or the internal ones but they come from our own big jar.